File: QytetetRuby/lib/tablero.rb
#///////////////////////////////////////////////////////////////////////////////
#// Autor: Alejandro Manzanares Lemus
#// 2º GII A1
#///////////////////////////////////////////////////////////////////////////////
module ModeloQytetet
  class Tablero
    
    def initialize()
      @casillas
      @carcel
      inicializar
    end

    attr_reader :casillas, :carcel
    
    def es_casilla_carcel(numero_casilla)
      escarcel = false
      if numero_casilla == @carcel.numero_casilla
        escarcel = true
      end
      
      return escarcel
    end
    def obtener_casilla_final(casilla, desplazamiento)
       tam = @casillas.length()
       num_casilla_final = (casilla.numero_casilla + desplazamiento) % tam
       @casillas[num_casilla_final]
    end     
    def obtener_casilla_numero(numero_casilla)
      @casillas[numero_casilla]

    end
    def to_s()
      "TABLERO:
Casillas: #{@casillas}
Carcel: #{@carcel}"
    end
    
    private
    
    def inicializar()
      @casillas = Array.new
      @casillas<< Casilla.new(0, TipoCasilla::SALIDA)
      @casillas<< Calle.new(1, TituloPropiedad.new("Calle Navajazo en el pecho", 
          500, 50, -0.1, 250, 150))
      @casillas<< Casilla.new(2,TipoCasilla::JUEZ)
      @casillas<< Calle.new(3, TituloPropiedad.new("Plaza de Maria y Juana",
          550, 55, 0.1, 300, 160))
      @casillas<< Calle.new(4, TituloPropiedad.new("Calle Lucia S. y diamantes",
          600, 60, 0.11, 315, 200))
      @casillas<< Casilla.new(5,TipoCasilla::SORPRESA)
      @casillas<< Calle.new(6, TituloPropiedad.new("Plaza Mediocre",
          600, 60, 0.11, 315, 210))
      @casillas<< Casilla.new(7,TipoCasilla::IMPUESTO)
      @casillas<< Calle.new(8, TituloPropiedad.new("Calle Mumotopia",
          500, 75, 0.2, 320, 270))
      @casillas<< Calle.new(9, TituloPropiedad.new("Callejon de la piltrafa spooky",
          650, 80,0.15, 320, 350))
      @casillas<< Casilla.new(10, TipoCasilla::CARCEL)
      @casillas<< Calle.new(11, TituloPropiedad.new("Calle Fantasma que recorre Europa",
          700, 80, 0.15, 350, 400))
      @casillas<< Casilla.new(12,TipoCasilla::SORPRESA)
      @casillas<< Calle.new(13, TituloPropiedad.new("Avenida Casinos",
          750, 85, 0.15, 400, 450))
      @casillas<< Calle.new(14, TituloPropiedad.new("Plaza Joestar",
          800, 85, 0.15, 415, 550))
      @casillas<< Casilla.new(15,TipoCasilla::PARKING)
      @casillas<< Calle.new(16, TituloPropiedad.new("Calle Iluminada",
          850 , 90, 0.15, 500, 600))
      @casillas<< Casilla.new(17,TipoCasilla::SORPRESA)
      @casillas<< Calle.new(18, TituloPropiedad.new("Avenida Ariedad",
          900, 95, 0.15, 600, 700))
      @casillas<< Calle.new(19, TituloPropiedad.new("Parque Illuminati",
          1000, 100, 0.2, 700, 900))
      @carcel = @casillas[10]
    end
  end
end

File: QytetetJava/src/modeloqytetet/Tablero.java
////////////////////////////////////////////////////////////////////////////////
// Autor: Alejandro Manzanares Lemus
// 2º GII A1
////////////////////////////////////////////////////////////////////////////////
package modeloqytetet;
import java.util.ArrayList;

public class Tablero {
    //Atributos
    private ArrayList<Casilla> casillas = new ArrayList<Casilla>();
    private Casilla carcel;

    //Constructor
    public Tablero() {
        inicializar();
    }
    
    //Metodos
    boolean esCasillaCarcel(int numeroCasilla){
        boolean escarcel = false;
        if(numeroCasilla == carcel.getNumeroCasilla()){
            escarcel = true;
        }
        return escarcel;
    }
    public Casilla getCarcel() {
        return carcel;
    }
    public ArrayList<Casilla> getCasillas() {
        return casillas;
    }
    private void inicializar() {
        casillas = new ArrayList<Casilla>();
        TituloPropiedad Navajazo = new TituloPropiedad(
                "Calle Navajazo en el pecho",
        500, 50, (float) -0.1, 250, 150);
        TituloPropiedad Mariajuana = new TituloPropiedad(
                "Plaza de Maria y Juana",
        550, 55, (float) 0.1, 300, 160);
        TituloPropiedad LuciaS = new TituloPropiedad(
                "Calle Lucia S. y diamantes",
        600, 60, (float) 0.11, 315, 200);
        TituloPropiedad Mediocre = new TituloPropiedad(
                "Plaza Mediocre",
        600, 60, (float) 0.11, 315, 210);
        TituloPropiedad Mumotopia = new TituloPropiedad(
                "Calle Mumotopia",
        500, 75, (float) 0.2, 320, 270);
        TituloPropiedad Piltrafa = new TituloPropiedad(
                "Callejon de la piltrafa spooky",
        650, 80,(float) 0.15, 320, 350);
        TituloPropiedad Fantasma = new TituloPropiedad(
                "Calle Fantasma que recorre Europa",
        700, 80, (float) 0.15, 350, 400);
        TituloPropiedad Casinos = new TituloPropiedad(
                "Avenida Casinos",
        750, 85, (float) 0.15, 400, 450);
        TituloPropiedad Joestar = new TituloPropiedad(
                "Plaza Joestar",
        800, 85, (float) 0.15, 415, 550);
        TituloPropiedad Iluminada = new TituloPropiedad(
                "Calle Iluminada",
        850 , 90, (float) 0.15, 500, 600);
        TituloPropiedad Ariedad = new TituloPropiedad(
                "Avenida Ariedad",
        900, 95, (float) 0.15, 600, 700);
        TituloPropiedad Illuminati = new TituloPropiedad(
                "Parque Illuminati",
        1000, 100, (float) 0.2, 700, 900);
        
        casillas.add(new OtraCasilla(0, TipoCasilla.SALIDA));
        casillas.add(new Calle(1, Navajazo));
        casillas.add(new OtraCasilla(2, TipoCasilla.JUEZ));
        casillas.add(new Calle(3, Mariajuana));
        casillas.add(new Calle(4, LuciaS));
        casillas.add(new OtraCasilla(5,TipoCasilla.SORPRESA));
        casillas.add(new Calle(6, Mediocre));
        casillas.add(new OtraCasilla(7,TipoCasilla.IMPUESTO));
        casillas.add(new Calle(8, Mumotopia));
        casillas.add(new Calle(9, Piltrafa));
        casillas.add(new OtraCasilla(10, TipoCasilla.CARCEL));
        casillas.add(new Calle(11, Fantasma));
        casillas.add(new OtraCasilla(12,TipoCasilla.SORPRESA));
        casillas.add(new Calle(13, Casinos));
        casillas.add(new Calle(14, Joestar));
        casillas.add(new OtraCasilla(15, TipoCasilla.PARKING));
        casillas.add(new Calle(16, Iluminada));
        casillas.add(new OtraCasilla(17,TipoCasilla.SORPRESA));
        casillas.add(new Calle(18, Ariedad));
        casillas.add(new Calle(19, Illuminati));
        
        carcel = casillas.get(10);
    }  
    Casilla obtenerCasillaFinal(Casilla casilla, int desplazamiento){
       int tam = casillas.size();
       int numCasillaFinal = (casilla.getNumeroCasilla()+desplazamiento)%tam;
       return casillas.get(numCasillaFinal);
    }
    Casilla obtenerCasillaNumero(int numeroCasilla){
       int numCasillaObtenida = numeroCasilla;
       return casillas.get(numCasillaObtenida);
    }
    @Override
    public String toString() {
        return "Tablero{" + "casillas=" + casillas + ", carcel=" + carcel + '}';
    }
}
